doc/plugins.dox

```diff
@@ -39,7 +40,7 @@ We will try to document as many functions and structs as possible.
 
 @section pluginsupport Plugin Support
 - @link howto Plugin HowTo @endlink - get started
-- @link pluginsymbols.c Plugin Symbols @endlink
+- @link legacy Legacy Plugin Entry Points and Transition to the Modern ones @endlink
```

Comment:
"Modern" is not quite the right word, current or new would be better

transitioning